backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/components/CassandraModule.java

```diff
@@ -21,8 +21,42 @@
 
 import java.util.List;
 
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.schemabuilder.CreateType;
+import com.google.common.collect.ImmutableList;
+
 public interface CassandraModule {
 
+    static CassandraModule forType(String name, CreateType statement) {
```

Comment:
After thinking about it, I don't think that these two methods are good enough from the caller perspective. I would rather implement a nice builder that would allow that : 
```
CassandraModule.builder().type().name("foo").statement(statement).build();
CassandraModule.builder().table().name("foo").statement(statement).build();
CassandraModule.builder()
.table().name("foo").statement(statement)
.table().name("bar").statement(statement)
.type().name("baz").statement(statement).build();
```
It would allow to never extend any class for describing table and type structure.
WDYT ?

Reply:
I think the staged builder is not required.

Ok for the normal builder, but it will take time to implement.

Reply:
> I think the staged builder is not required.

Maybe you can explain me why you think so ? What about giving me an example of the builder call like I did so that other people can actually evaluate your proposition ?

Reply:
Well, because on top of each table we do have a (hopefully tested) DAO

To run this DAO fast we need a module with only that table.

Which means that with our design definitions, if we want to enforce fast run, 1 table = 1 module (to allow fast DAO tests). That's all this PR is about.

Now investing build time, to write a builder that breaks that principle (and in the very same PR) do not seem a great idea to me.

Reply:
I don't complain that it's `more work` but rather `more work for nothing` and gave a rather clear explanation about it.

Now if I do encounter a place where I need it, I introduce it. If I don't (accordingly to this changeset) why would I?

Reply:
> gave a rather clear explanation about it.

You probably think you did but you can see that I'm still answering questions or expect more details to my questions, so no, you are not doing explaining, sorry.

Reply:
Anyway, now the stage builder is available ;-)